packages/edit-widgets/src/components/customizer-edit-widgets-initializer/sync-customizer.js

```diff
@@ -78,9 +78,19 @@ const updateSettingInputValue = throttle( ( nextWidgetAreas ) => {
 
 // Check that all the necessary globals are present.
 if ( window.wp && window.wp.customize && window.wp.data ) {
+	let ran = false;
 	// Wait for the Customizer to finish bootstrapping.
 	window.wp.customize.bind( 'ready', () =>
```

Comment:
Can't we defer all this initialization until the Widget Blocks section is opened for the first time?

Reply:
I couldn't find any relevant events in the API. We could manually add a listener to the section title element. Should I explore that?

Reply:
I don't think that's a blocker but it would be nice to explore that, as it provides an extra layer to avoid problems for users that never opened the section.

Reply:
I've given this more thought.

>Imaging a scenario where some plugin disables/removes the filters to add the new widget block section, in this case, we should not execute this code.

It wouldn't conflict or throw any errors, because all the element selecting and referencing happens inside a try block or has if guards.

I think we should avoid the complexity of selecting yet another element with a css selector if possible, specially the one that boots everything up. I can see a scenario where the click handler is not attached because the section is not rendered in the UI yet, but it wouldn't have been a problem with the current code, because it would have just started syncing once all the necessary elements are rendered.